geoengine/colorizer.py

```diff
@@ -1,128 +1,194 @@
-                    "'hot_r', 'hsv_r', 'jet_r', 'nipy_spectral_r', 'ocean_r', 'pink_r', 'prism_r', "
-                    "'rainbow_r', 'seismic_r', 'spring_r', 'summer_r', 'terrain_r', 'winter_r', 'Accent_r', "
-                    "'Dark2_r', 'Paired_r', 'Pastel1_r', 'Pastel2_r', 'Set1_r', 'Set2_r', 'Set3_r', "
-                    "'tab10_r', 'tab20_r', 'tab20b_r', 'tab20c_r']")
-
         if n_steps < 2:
             raise ValueError(f"n_steps must be greater than or equal to 2, got {n_steps} instead.")
         if min_max[1] <= min_max[0]:
             raise ValueError(f"min_max[1] must be greater than min_max[0], got {min_max[1]} and {min_max[0]}.")
-        if len(default_color) != 4:
-            raise ValueError(f"defaultColor must be a tuple of length 4, got {len(default_color)} instead.")
+        if len(over_color) != 4:
+            raise ValueError(f"overColor must be a tuple of length 4, got {len(over_color)} instead.")
+        if len(under_color) != 4:
+            raise ValueError(f"underColor must be a tuple of length 4, got {len(under_color)} instead.")
         if len(no_data_color) != 4:
             raise ValueError(f"noDataColor must be a tuple of length 4, got {len(no_data_color)} instead.")
         if not all(0 <= elem < 256 for elem in no_data_color):
             raise ValueError(f"noDataColor must be a RGBA color specification, got {no_data_color} instead.")
-        if not all(0 <= elem < 256 for elem in default_color):
-            raise ValueError(f"defaultColor must be a RGBA color specification, got {default_color} instead.")
+        if not all(0 <= elem < 256 for elem in over_color):
+            raise ValueError(f"overColor must be a RGBA color specification, got {over_color} instead.")
+        if not all(0 <= elem < 256 for elem in under_color):
+            raise ValueError(f"underColor must be a RGBA color specification, got {under_color} instead.")
 
-        # get the map, and transform it to [0,255] values
-        colormap = [
-            (int(x[0]), int(x[1]), int(x[2]), int(x[3])) for x in ScalarMappable(cmap=map_name).to_rgba(
-                np.linspace(min_max[0], min_max[1], n_steps), bytes=True)]
+        # get the map, and transform it to a list of (uint8) rgba values
+        list_of_rgba_colors: List[npt.NDArray[np.uint8]] = ScalarMappable(cmap=color_map).to_rgba(
+            np.linspace(min_max[0], min_max[1], n_steps), bytes=True)
 
         # if you want to remap the colors, you can do it here (e.g. cutting of the most extreme colors)
-        value_bounds = [
-            int(x) for x in np.linspace(min_max[0], min_max[1], n_steps)
+        values_of_breakpoints: List[float] = np.linspace(min_max[0], min_max[1], n_steps).tolist()
+
+        # generate color map steps for geoengine
+        breakpoints: List[ColorBreakpoint] = [
+            ColorBreakpoint(
+                color=cast(Rgba, tuple(color.tolist())), value=value
```

Comment:
i think it is more intuitive to just create a new Rgba tuple as (color[0], color[1], ...)

Reply:
Good idea, I had to add the `int()` wrappings though, because `color` contains `np.uint8` in this case.